tests/infra_test.go

```diff
@@ -252,24 +253,26 @@ var _ = Describe("[Serial]Infrastructure", func() {
 				Expect(err).ShouldNot(HaveOccurred(), "failed listing kubevirt pods")
 				Expect(len(pods.Items)).To(BeNumerically(">", 0), "no kubevirt pods found")
 
-				By("finding all schedulable nodes")
-				schedulableNodesList := tests.GetAllSchedulableNodes(virtClient)
-				schedulableNodes := map[string]*k8sv1.Node{}
-				for _, node := range schedulableNodesList.Items {
-					schedulableNodes[node.Name] = node.DeepCopy()
-				}
-
 				By("selecting one compute only node that runs kubevirt components")
 				// master nodes should never have the CriticalAddonsOnly taint because core components might not
 				// tolerate this taint because it is meant to be used on compute nodes only. If we set this taint
 				// on a master node, we risk in breaking the test cluster.
-				for _, pod := range pods.Items {
-					node := schedulableNodes[pod.Spec.NodeName]
-					if _, isMaster := node.Labels["node-role.kubernetes.io/master"]; isMaster {
-						continue
+				Eventually(func() *k8sv1.Node {
+					schedulableNodesList := tests.GetAllSchedulableNodes(virtClient)
+					for _, node := range schedulableNodesList.Items {
+						if _, isMaster := node.Labels["node-role.kubernetes.io/master"]; isMaster {
+							continue
+						}
+						for _, pod := range pods.Items {
+							if pod.Spec.NodeName == node.Name {
+								selectedNode = node.DeepCopy()
+								return selectedNode
```

Comment:
in the code prior to your change, selectedNode will be given to the node that hold the last pod in the loop (since we're iterating over the pods and set selectedNode each time).
in your changed code, selectedNode will be given to the node that hold the first pod in the loop.
This could have the consistent outcome only if there is 1 pod in the list..

can you explain why this change is ok?

Reply:
because we just need a node which isn't the master node, the previous code just continued iterating for no reason at all.